src/benchmarks/micro/libraries/Microsoft.Extensions.Primitives/StringValuesBenchmark.cs

```diff
@@ -0,0 +1,238 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Running;
+using MicroBenchmarks;
+
+namespace Microsoft.Extensions.Primitives.Performance
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class StringValuesBenchmark
+    {
+        const int Iterations = 40;
+
+        private readonly string _string = "Hello world!";
+        private readonly string[] _stringArray = new[] { "Hello", "world", "!" };
+        private readonly StringValues _stringBased = new StringValues("Hello world!");
+        private readonly StringValues _arrayBased = new StringValues(new[] { "Hello", "world", "!" });
+
+#pragma warning disable CS0414
+        private HeaderReferences _headers;
+#pragma warning restore CS0414
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void ClearHeaders()
+        {
+            for (var i = 0; i < Iterations; i++)
+            {
+                ClearHeaders_Body();
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void ClearHeaders_Body()
+        {
+            _headers = default;
+        }
```

Comment:
this benchmark should rather not be a part of the performance repo: it's measuring the cost of nulling a big struct used internally by ASP.NET Core. Imho we should remove it.
```suggestion
```